src/controller/python/chip/ChipDeviceCtrl.py

```diff
@@ -196,7 +196,7 @@ def ConnectIP(self, ipaddr, setupPinCode, nodeid):
         )
 
     def ResolveNode(self, fabricid, nodeid):
-        return self._ChipStack.CallAsync(
+        return self._ChipStack.Call(
```

Comment:
This will cause that `resolve` will not wait for the response before returning to the prompt, no? That would be undesired as "resolve" command expects that when `ResolveNode` exits an address of the device is already resolved. `CallAsync` starts an asynchronous operation and returns to the prompt only when `HandleAddressUpdateComplete` is called. The question is why `HandleAddressUpdateComplete` it not called in your case.

Reply:
>That would be undesired as "resolve" command expects that when ResolveNode exits an address of the device is already resolved.

This seems right.

So, is there a timeout to the resolve command (within the code)? So that it can terminate in case the operation isn't successful and the python controller prompt does not remain in the frozen/hanged state.

Reply:
@Damian-Nordic you are right. `HandleAddressComplete` is called in my case too the problem is with the wrong enum used.

Reply:
@dhrishi python controller remained frozen due to undefined use of enum. A new change would resolve the issue with no need for a timeout.

Reply:
Thanks!